src/statistical/unigram.jl

```diff
@@ -32,31 +30,24 @@ end
 use to initialize the SentencePieceModel by providing `vocab filepath`
 """    
 function load(path; unk_token="<unk>")
-    vocab = readlines(path)
-    vocabnew = split.(vocab, "\t")
-    voc = String[]
-    logprob = Float64[]
-    for i in 1:length(vocab)
-        vocab1 = vocabnew[i][1]
-        vocab1 = replace(vocab1, "▁"=>"_")
-        push!(voc, vocab1)
-        logp = vocabnew[i][2]
+    vocab_path = readlines(path)
+    vocabnlogp = split.(vocab_path, "\t")
+    vocab_map = Dict{String, Tuple{Float64, Int}}()
+    for i in 1:length(vocabnlogp)
+        vocab = vocabnlogp[i][1]
+        logp = vocabnlogp[i][2]
         logp =  parse(Float64, logp)
-        push!(logprob, logp)    
+        push!(vocab_map, vocab => (logp, i))    
     end
-    unk_id = findall(x->x==unk_token, voc)
-    length(unk_id) == 0 && throw(UndefVarError(:unk_token)) 
-    spm = SentencePieceModel(voc, logprob, unk_id[1])
+    if haskey(vocab_map, unk_token)
+        unk_id = vocab_map[unk_token][2]
+    else
+        throw(UndefVarError(:unk_token))  
```

Comment:
I think it is confusing to use `UndefVarError` for this. It should only really be thrown by the compiler.
 
```suggestion
        throw(DomainError(unk_token, "Unknown token is not in the vocabulary")
```